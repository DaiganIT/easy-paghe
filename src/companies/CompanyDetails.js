import React from 'react';
import { withStyles, Paper, Typography, TextField, Grid } from '@material-ui/core';
import CompanyBases from './CompanyBases';
import CompanyBase from './CompanyBase';

const styles = {
	paper: {
		padding: '1em',
		marginBottom: '1em',
	},
	miniTitle: {
		marginBottom: '1em',
	},
	textField: {
		marginRight: '1em',
		marginBottom: '1em',
	}
}

function CompanyDetails({ classes, isSaving, company, updateField, updateBaseField, selectedBaseIndex }) {
	return (
		<Grid container spacing={24}>
			<Grid item lg={6} xs={12}>
				<Paper className={classes.paper}>
					<Typography variant="title" className={classes.miniTitle}>
						Dettagli azienda
			</Typography>
					<TextField
						variant="outlined"
						label="Nome"
						fullWidth
						className={classes.textField}
						disabled={isSaving}
						value={company.name}
						onChange={(e) => updateField('name', e.target.value)}
					/>
					<TextField
						variant="outlined"
						label="Codice Fiscale"
						fullWidth
						className={classes.textField}
						disabled={isSaving}
						value={company.fiscalCode}
						onChange={(e) => updateField('fiscalCode', e.target.value)}
					/>
					<TextField
						variant="outlined"
						label="Partita IVA"
						fullWidth
						className={classes.textField}
						disabled={isSaving}
						value={company.ivaCode}
						onChange={(e) => updateField('ivaCode', e.target.value)}
					/>
					<TextField
						variant="outlined"
						label="Codice INPS"
						fullWidth
						className={classes.textField}
						disabled={isSaving}
						value={company.inpsRegistrationNumber}
						onChange={(e) => updateField('inpsRegistrationNumber', e.target.value)}
					/>
					<TextField
						variant="outlined"
						label="Codice INAIL"
						fullWidth
						className={classes.textField}
						disabled={isSaving}
						value={company.inailRegistrationNumber}
						onChange={(e) => updateField('inailRegistrationNumber', e.target.value)}
					/>
				</Paper>
			</Grid>
			<Grid item lg={6} xs={12}>
				<CompanyBases bases={company.bases} />
			</Grid>
			<Grid item lg={6} xs={12}>
				<CompanyBase base={company.bases[selectedBaseIndex]} updateBaseField={updateBaseField} index={selectedBaseIndex}/>
			</Grid>
		</Grid>
	);
}

export default withStyles(styles)(CompanyDetails);