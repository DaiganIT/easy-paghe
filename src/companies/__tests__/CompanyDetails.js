import React from 'react';
import {
  render,
  cleanup,
  fireEvent,
} from 'react-testing-library';
import 'jest-dom/extend-expect';
import CompanyDetails from '../CompanyDetails';

afterEach(cleanup)

const defaultProps = {
  isSaving: false,
  company: {
    name: 'The name',
    fiscalCode: 'TheFiscalCode',
    ivaCode: '5645',
    inpsRegistrationNumber: '54756',
    inailRegistrationNumber: '46854695',
  },
  updateField: jest.fn(),
  errors: {}
}

test('Company Details structure snapshot is correct', () => {
  const { container } = render(<CompanyDetails {...defaultProps} />);
  expect(container.firstChild).toMatchSnapshot();
});

const testCases = [
  {
    name: 'name',
    inputId: 'name'
  },
  {
    name: 'fiscalCode',
    inputId: 'fiscal-code'
  },
  {
    name: 'ivaCode',
    inputId: 'iva-code',
    isNumeric: true
  },
  {
    name: 'inpsRegistrationNumber',
    inputId: 'inps',
    isNumeric: true
  },
  {
    name: 'inailRegistrationNumber',
    inputId: 'inail',
    isNumeric: true
  },
]

testCases.forEach(testCase => {
  afterEach(jest.clearAllMocks);

  test(`Company Details ${testCase.name} input is rendered`, () => {
    const { container } = render(<CompanyDetails {...defaultProps} />);

    expect(container.querySelector(`input#company-${testCase.inputId}`).value).toBe(defaultProps.company[testCase.name]);
  });

  test(`Company Details ${testCase.name} input change is catched`, () => {
    const { container } = render(<CompanyDetails {...defaultProps} />);

    fireEvent.change(container.querySelector(`input#company-${testCase.inputId}`), { target: { value: testCase.isNumeric ? '54556' : 'Changed' } });
    expect(defaultProps.updateField).toHaveBeenCalledWith(testCase.name, testCase.isNumeric ? '54556' : 'Changed');
  });
});

test('Company Details inputs are disabled when isSaving is true', () => {
  const props = Object.assign({}, defaultProps, { isSaving: true });
  const { container } = render(<CompanyDetails {...props} />);

  const allInputs = container.querySelectorAll('input');
  for(const input of allInputs) {
    expect(input).toBeDisabled();
  }
});

const testCases2 = [
  {
    name: 'name',
    inputId: 'name'
  },
  {
    name: 'fiscalCode',
    inputId: 'fiscal-code'
  },
  {
    name: 'ivaCode',
    inputId: 'iva-code',
  },
  {
    name: 'inpsRegistrationNumber',
    inputId: 'inps',
  },
  {
    name: 'inailRegistrationNumber',
    inputId: 'inail',
  },
]

testCases2.forEach(testCase => {
  test(`Company Details ${testCase.name} input is rendered with errors`, () => {
    const props = Object.assign({}, defaultProps, { errors: { [testCase.name]: ['error'] }});
    const { container } = render(<CompanyDetails {...props} />);

    expect(container.querySelector(`label[for=company-${testCase.inputId}]`).className).toContain('MuiFormLabel-error');
  });
});