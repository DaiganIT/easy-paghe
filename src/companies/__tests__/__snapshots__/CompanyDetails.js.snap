// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company Details structure snapshot is correct 1`] = `
<div
  class="MuiGrid-container-2 MuiGrid-spacing-xs-16-25"
>
  <div
    class="MuiGrid-item-3 MuiGrid-grid-xs-12-42"
  >
    <div
      class="MuiFormControl-root-99 MuiFormControl-fullWidth-102 CompanyDetails-textField-1"
    >
      <label
        class="MuiFormLabel-root-114 MuiFormLabel-filled-118 MuiFormLabel-required-119 MuiInputLabel-required-107 MuiInputLabel-root-103 MuiInputLabel-formControl-108 MuiInputLabel-animated-111 MuiInputLabel-shrink-110 MuiInputLabel-outlined-113"
        data-shrink="true"
        for="company-name"
      >
        Nome
        <span
          class="MuiFormLabel-asterisk-120"
        >
           *
        </span>
      </label>
      <div
        class="MuiInputBase-root-134 MuiOutlinedInput-root-121 MuiInputBase-fullWidth-143 MuiInputBase-formControl-135"
      >
        <fieldset
          aria-hidden="true"
          class="MuiPrivateNotchedOutline-root-151 MuiOutlinedInput-notchedOutline-128"
          style="padding-left: 8px;"
        >
          <legend
            class="MuiPrivateNotchedOutline-legend-152"
            style="width: 0px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
        <input
          aria-invalid="false"
          class="MuiInputBase-input-144 MuiOutlinedInput-input-129"
          id="company-name"
          required=""
          type="text"
          value="The name"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-item-3 MuiGrid-grid-xs-6-36"
  >
    <div
      class="MuiFormControl-root-99 MuiFormControl-fullWidth-102 CompanyDetails-textField-1"
    >
      <label
        class="MuiFormLabel-root-114 MuiFormLabel-filled-118 MuiInputLabel-root-103 MuiInputLabel-formControl-108 MuiInputLabel-animated-111 MuiInputLabel-shrink-110 MuiInputLabel-outlined-113"
        data-shrink="true"
        for="company-fiscal-code"
      >
        Codice Fiscale
      </label>
      <div
        class="MuiInputBase-root-134 MuiOutlinedInput-root-121 MuiInputBase-fullWidth-143 MuiInputBase-formControl-135"
      >
        <fieldset
          aria-hidden="true"
          class="MuiPrivateNotchedOutline-root-151 MuiOutlinedInput-notchedOutline-128"
          style="padding-left: 8px;"
        >
          <legend
            class="MuiPrivateNotchedOutline-legend-152"
            style="width: 0px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
        <input
          aria-invalid="false"
          class="MuiInputBase-input-144 MuiOutlinedInput-input-129"
          id="company-fiscal-code"
          type="text"
          value="TheFiscalCode"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-item-3 MuiGrid-grid-xs-6-36"
  >
    <div
      class="MuiFormControl-root-99 MuiFormControl-fullWidth-102 CompanyDetails-textField-1"
    >
      <label
        class="MuiFormLabel-root-114 MuiFormLabel-filled-118 MuiInputLabel-root-103 MuiInputLabel-formControl-108 MuiInputLabel-animated-111 MuiInputLabel-shrink-110 MuiInputLabel-outlined-113"
        data-shrink="true"
        for="company-iva-code"
      >
        Partita IVA
      </label>
      <div
        class="MuiInputBase-root-134 MuiOutlinedInput-root-121 MuiInputBase-fullWidth-143 MuiInputBase-formControl-135"
      >
        <fieldset
          aria-hidden="true"
          class="MuiPrivateNotchedOutline-root-151 MuiOutlinedInput-notchedOutline-128"
          style="padding-left: 8px;"
        >
          <legend
            class="MuiPrivateNotchedOutline-legend-152"
            style="width: 0px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
        <input
          aria-invalid="false"
          class="MuiInputBase-input-144 MuiOutlinedInput-input-129 MuiInputBase-inputType-147"
          id="company-iva-code"
          type="number"
          value="5645"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-item-3 MuiGrid-grid-xs-6-36"
  >
    <div
      class="MuiFormControl-root-99 MuiFormControl-fullWidth-102 CompanyDetails-textField-1"
    >
      <label
        class="MuiFormLabel-root-114 MuiFormLabel-filled-118 MuiInputLabel-root-103 MuiInputLabel-formControl-108 MuiInputLabel-animated-111 MuiInputLabel-shrink-110 MuiInputLabel-outlined-113"
        data-shrink="true"
        for="company-inps"
      >
        Codice INPS
      </label>
      <div
        class="MuiInputBase-root-134 MuiOutlinedInput-root-121 MuiInputBase-fullWidth-143 MuiInputBase-formControl-135"
      >
        <fieldset
          aria-hidden="true"
          class="MuiPrivateNotchedOutline-root-151 MuiOutlinedInput-notchedOutline-128"
          style="padding-left: 8px;"
        >
          <legend
            class="MuiPrivateNotchedOutline-legend-152"
            style="width: 0px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
        <input
          aria-invalid="false"
          class="MuiInputBase-input-144 MuiOutlinedInput-input-129 MuiInputBase-inputType-147"
          id="company-inps"
          type="number"
          value="54756"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-item-3 MuiGrid-grid-xs-6-36"
  >
    <div
      class="MuiFormControl-root-99 MuiFormControl-fullWidth-102 CompanyDetails-textField-1"
    >
      <label
        class="MuiFormLabel-root-114 MuiFormLabel-filled-118 MuiInputLabel-root-103 MuiInputLabel-formControl-108 MuiInputLabel-animated-111 MuiInputLabel-shrink-110 MuiInputLabel-outlined-113"
        data-shrink="true"
        for="company-inail"
      >
        Codice INAIL
      </label>
      <div
        class="MuiInputBase-root-134 MuiOutlinedInput-root-121 MuiInputBase-fullWidth-143 MuiInputBase-formControl-135"
      >
        <fieldset
          aria-hidden="true"
          class="MuiPrivateNotchedOutline-root-151 MuiOutlinedInput-notchedOutline-128"
          style="padding-left: 8px;"
        >
          <legend
            class="MuiPrivateNotchedOutline-legend-152"
            style="width: 0px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
        <input
          aria-invalid="false"
          class="MuiInputBase-input-144 MuiOutlinedInput-input-129 MuiInputBase-inputType-147"
          id="company-inail"
          type="number"
          value="46854695"
        />
      </div>
    </div>
  </div>
</div>
`;
